src/gnark_backend_wrapper/plonk/mod.rs

```diff
@@ -0,0 +1,252 @@
+        .map_err(|e| GnarkBackendError::SerializeCircuitError(e.to_string()))?;
+    let acir_go_string = GoString::try_from(&acir_c_str)?;
+
+    let values_json = serde_json::to_string(&circuit)
+        .map_err(|e| GnarkBackendError::SerializeCircuitError(e.to_string()))?;
+    let values_c_str = CString::new(values_json)
+        .map_err(|e| GnarkBackendError::SerializeCircuitError(e.to_string()))?;
+    let values_go_string = GoString::try_from(&values_c_str)?;
+
+    let serialized_proof = String::from_utf8(proof.to_vec())
+        .map_err(|e| GnarkBackendError::SerializeProofError(e.to_string()))?;
+    let c_str = CString::new(serialized_proof)
+        .map_err(|e| GnarkBackendError::SerializeProofError(e.to_string()))?;
+    let go_string_proof = GoString::try_from(&c_str)?;
+
+    let result =
+        unsafe { PlonkVerifyWithMeta(circuit_go_string, values_go_string, go_string_proof) };
+    match result {
+        0 => Ok(false),
+        1 => Ok(true),
+        _ => Err(GnarkBackendError::VerifyInvalidBoolError),
+    }
+}
+
+pub fn verify_with_vk(
+    circuit: &acvm::Circuit,
+    proof: &[u8],
+    public_inputs: &[acvm::FieldElement],
+    verifying_key: &[u8],
+) -> Result<bool, GnarkBackendError> {
+    // Serialize to json and then convert to GoString
+    let acir_json = serde_json::to_string(&circuit)
+        .map_err(|e| GnarkBackendError::SerializeCircuitError(e.to_string()))?;
+    let acir_c_str = CString::new(acir_json)
+        .map_err(|e| GnarkBackendError::SerializeCircuitError(e.to_string()))?;
+    let acir_go_string = GoString::try_from(&acir_c_str)?;
+
+    let values_json = serde_json::to_string(&circuit)
+        .map_err(|e| GnarkBackendError::SerializeCircuitError(e.to_string()))?;
+    let values_c_str = CString::new(values_json)
+        .map_err(|e| GnarkBackendError::SerializeCircuitError(e.to_string()))?;
+    let values_go_string = GoString::try_from(&values_c_str)?;
+
+    let proof_serialized = hex::encode(proof);
+    let proof_c_str = CString::new(proof_serialized)
+        .map_err(|e| GnarkBackendError::SerializeProofError(e.to_string()))?;
+    let proof_go_string = GoString::try_from(&proof_c_str)?;
+
+    let verifying_key_serialized = hex::encode(verifying_key);
+    let verifying_key_c_str = CString::new(verifying_key_serialized)
+        .map_err(|e| GnarkBackendError::SerializeKeyError(e.to_string()))?;
+    let verifying_key_go_string = GoString::try_from(&verifying_key_c_str)?;
+
+    let verifies = unsafe {
+        PlonkVerifyWithVK(
+            circuit_go_string,
+            values_go_string,
+            proof_go_string,
+            verifying_key_go_string,
+        )
+    };
+    match verifies {
+        0 => Ok(false),
+        1 => Ok(true),
+        _ => Err(GnarkBackendError::VerifyInvalidBoolError),
+    }
+}
+
+pub fn get_exact_circuit_size(circuit: &acvm::Circuit) -> Result<u32, GnarkBackendError> {
+    let size: u32 = RawR1CS::num_constraints(circuit)?
```

Comment:
Maybe we could expose the `num_constraints` function so both backends could use it (instead of using a struct of another backend here)